crates/core/src/operations/merge/mod.rs

```diff
@@ -997,7 +997,7 @@ async fn execute(
 
     let source_schema = source.schema();
     let target_schema = target.schema();
-    let join_schema_df = build_join_schema(source_schema, &target_schema, &JoinType::Full)?;
+    let join_schema_df = build_join_schema(source_schema, target_schema, &JoinType::Full)?;
```

Comment:
Why is this not a reference anymore?

Reply:
That's a clippy warning, because `target.schema()` already [returns a reference](https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.LogicalPlan.html#method.schema), so that double reference was being automatically dereferenced.